portal-impl/src/com/liferay/portal/events/ServicePreAction.java

```diff
@@ -908,11 +908,11 @@ private ThemeDisplay _initThemeDisplay(
 		}
 
 		if (!user.isActive()) {
-			user = company.getGuestUser();
-
-			if (realUser == null) {
+			if (realUserId == user.getUserId()) {
```

Comment:
@MirnaGama, doing some tests, I can see that the `realUser` is never `null` so the conditional check you used, the `realUserId == user.getUserId()` should be ok here. This solves the problem that is described in the bug ticket...